api/src/pcapi/core/educational/api/institution.py

```diff
@@ -57,3 +60,36 @@ def update_educational_institution_data(
     for key, value in institution_data.items():
         setattr(educational_institution, key, value)
     return educational_institution
+
+
+def compare_adage_and_institution(ansco: str) -> None:
+    institutions = list(educational_repository.get_all_educational_institutions()[0])
+    page = 1
+    adage_institutions_total = []
+    adage_institutions = None
+    while not adage_institutions:
+        adage_institutions = list(adage_client.get_adage_educational_institutions(ansco=ansco, page=page))
+        adage_institutions_total.extend(adage_institutions)
+        page += 1
```

Comment:
ton algo ne passera que sur la premiere page. 

je te propose un truc comme ça (a tester)

```suggestion
    page = 1
    while adage_institutions := list(adage_client.get_adage_educational_institutions(ansco=ansco, page=page)):
        adage_institutions_total.extend(adage_institutions)
        page += 1
```

de plus je pense que ce bout de code devrait être dans adage_backends/adage.py:get_adage_educational_institutions